<?xml version="1.0"?>

<launch>
  <arg name="robot_name" default="my_robot"/>
  <group ns="$(arg robot_name)">
    <group ns="phycon">
      <node name="wheel0" pkg="srs004host" type="phycon_wheel_actual" >
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
        <remap from="CANLink_in" to="/$(arg robot_name)/comport/CANLink_in"/>
        <param name="CAN_CH" value="A"/>
        <param name="CAN_ID" value="1"/>
        <param name="PPR" value="-397"/>
      </node>
      <node name="wheel1" pkg="srs004host" type="phycon_wheel_actual" >
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
        <remap from="CANLink_in" to="/$(arg robot_name)/comport/CANLink_in"/>
        <param name="CAN_CH" value="A"/>
        <param name="CAN_ID" value="2"/>
        <param name="PPR" value="-397"/>
      </node>
      <node name="wheel2" pkg="srs004host" type="phycon_wheel_actual" >
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
	<remap from="CANLink_in" to="/$(arg robot_name)/comport/CANLink_in"/>
        <param name="CAN_CH" value="A"/>
        <param name="CAN_ID" value="3"/>
        <param name="PPR" value="-397"/>
      </node>

      <node name="gun" pkg="srs004host" type="phycon_gun_actual" >
        <remap from="twist" to="gun/twist"/>
        <remap from="shot"  to="gun/shot"/>
        <remap from="laser" to="gun/laser"/>
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
        <param name="CAN_CH" value="A"/>
        <param name="CAN_ID" value="0"/>
        <param name="y_ajust" value="0.055"/>
        <param name="z_ajust" value="-0.045"/>

      </node>

      <node name="hit0" pkg="srs004host" type="phycon_hitsensor_actual" >
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
        <remap from="CANLink_in" to="/$(arg robot_name)/comport/CANLink_in"/>
        <param name="CAN_CH" value="A"/>
	<param name="CAN_ID" value="5"/>
      </node>

      <node name="master" pkg="srs004host" type="phycon_master_actual" output="screen">
        <remap from="CANLink_out" to="/$(arg robot_name)/comport/CANLink_out"/>
        <remap from="CANLink_in" to="/$(arg robot_name)/comport/CANLink_in"/>
        <param name="CAN_CH" value="S"/>
        <param name="CAN_ID" value="1"/>
      </node>


    </group>
    <group ns="comport">
      <node name="canlink_to_seriallink" pkg="srs_common" type="canlink_to_seriallink" />
      <node name="seriallink_decode" pkg="srs_common" type="seriallink_decode" />
<!--
      <node name="serialport" pkg="srs_common" type="serialport" />
-->
      <node name="seriallink_encode" pkg="srs_common" type="seriallink_encode" />
      <node name="seriallink_to_canlink" pkg="srs_common" type="seriallink_to_canlink" />
    </group>

  </group>
</launch>

